consensus/context_deserialize/src/milhouse.rs

```diff
@@ -0,0 +1,45 @@
+use crate::ContextDeserialize;
+use milhouse::{List, Value, Vector};
+use serde::de::Deserializer;
+use ssz_types::typenum::Unsigned;
+
+impl<'de, C, T, N> ContextDeserialize<'de, C> for List<T, N>
+where
+    T: ContextDeserialize<'de, C> + Value,
+    N: Unsigned,
+    C: Clone,
+{
+    fn context_deserialize<D>(deserializer: D, context: C) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        // First deserialize as a Vec.
+        // This is not the most efficient implementation as it allocates a temporary Vec. In future
+        // we could write a more performant implementation using `List::builder()`.
+        let vec = Vec::<T>::context_deserialize(deserializer, context)?;
```

Comment:
I simplified this impl in this commit:

- https://github.com/sigp/lighthouse/pull/7372/commits/14c3ce6364d8ddb55044662d6da898501b338ca2

It's the same efficiency as what you had + more obviously correct IMO. If we want a faster impl in future, the option is there.

Reply:
Awesome! Thank you :)